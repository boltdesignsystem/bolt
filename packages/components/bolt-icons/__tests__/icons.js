import { render, stopServer, html } from '../../../testing/testing-helpers';

const timeout = 60000;

describe('<bolt-icon> Component', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic usage', async () => {
    const results = await render('@bolt-components-icon/icon.twig', {
      name: 'add-open',
      background: 'square',
      size: 'medium',
      color: 'teal',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-icon> with Shadow DOM renders', async function() {
    const defaultIconShadowRoot = await page.evaluate(async () => {
      const icon = document.createElement('bolt-icon');
      icon.setAttribute('name', 'add-open');
      icon.setAttribute('background', 'circle');
      icon.setAttribute('size', 'large');
      document.body.appendChild(icon);
      await customElements.whenDefined('bolt-icon');
      await icon.updateComplete;
      return icon.renderRoot.innerHTML;
    });

    const defaultIconOuter = await page.evaluate(async () => {
      const icon = document.createElement('bolt-icon');
      icon.setAttribute('name', 'add-open');
      icon.setAttribute('background', 'circle');
      icon.setAttribute('size', 'large');
      document.body.appendChild(icon);
      await customElements.whenDefined('bolt-icon');
      await icon.updateComplete;
      return icon.outerHTML;
    });

    const renderedShadowDomHTML = await html(defaultIconShadowRoot);
    const renderedHTML = await html(defaultIconOuter);

    expect(renderedShadowDomHTML).toMatchSnapshot();
    expect(renderedHTML).toMatchSnapshot();
  });

  test('Additional <bolt-icon> with Shadow DOM renders', async function() {
    const defaultIconShadowRoot = await page.evaluate(async () => {
      const icon = document.createElement('bolt-icon');
      icon.setAttribute('name', 'yeti');
      icon.setAttribute('background', 'circle');
      icon.setAttribute('size', 'large');
      document.body.appendChild(icon);
      await customElements.whenDefined('bolt-icon');
      await icon.updateComplete;
      return icon.renderRoot.innerHTML;
    });

    const defaultIconOuter = await page.evaluate(() => {
      const icon = document.createElement('bolt-icon');
      icon.setAttribute('name', 'yeti');
      icon.setAttribute('background', 'circle');
      icon.setAttribute('size', 'large');
      document.body.appendChild(icon);
      icon.updateComplete;
      return icon.outerHTML;
    });

    const renderedShadowDomHTML = await html(defaultIconShadowRoot);
    const renderedHTML = await html(defaultIconOuter);

    expect(renderedShadowDomHTML).toMatchSnapshot();
    expect(renderedHTML).toMatchSnapshot();
  });
});
