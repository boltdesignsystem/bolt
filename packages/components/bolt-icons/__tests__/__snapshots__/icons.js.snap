// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-icon> Component Additional <bolt-icon> with Shadow DOM renders 1`] = `
<span
  class="c-bolt-icon c-bolt-icon--large c-bolt-icon--yeti has-background has-circle-background c-bolt-icon--auto"
  style=""
>
  
        
              
  <span
    class="c-bolt-icon__background-shape c-bolt-icon__background-shape--circle"
    style=""
  />
  
            
        
  <svg
    class="c-bolt-icon__icon c-bolt-icon__icon--large"
    style=""
  >
    
          
    <use
      href="#yeti"
      style=""
    />
    
        
  </svg>
  
      
</span>
`;

exports[`<bolt-icon> Component Additional <bolt-icon> with Shadow DOM renders 2`] = `
<bolt-icon
  background="circle"
  name="yeti"
  size="large"
  style=""
/>
`;

exports[`<bolt-icon> Component Default <bolt-icon> with Shadow DOM renders 1`] = `
<span
  class="c-bolt-icon c-bolt-icon--large c-bolt-icon--add-open has-background has-circle-background c-bolt-icon--auto"
  style=""
>
  
        
              
  <span
    class="c-bolt-icon__background-shape c-bolt-icon__background-shape--circle"
    style=""
  />
  
            
        
  <svg
    class="c-bolt-icon__icon c-bolt-icon__icon--large"
    style=""
  >
    
          
    <use
      href="#add-open"
      style=""
    />
    
        
  </svg>
  
      
</span>
`;

exports[`<bolt-icon> Component Default <bolt-icon> with Shadow DOM renders 2`] = `
<bolt-icon
  background="circle"
  name="add-open"
  size="large"
  style=""
>
  <!---->
  
      
      
  <span
    class="c-bolt-icon c-bolt-icon--large c-bolt-icon--add-open has-background has-circle-background c-bolt-icon--auto"
    style=""
  >
    
        
              
    <span
      class="c-bolt-icon__background-shape c-bolt-icon__background-shape--circle"
      style=""
    />
    
            
        
    <svg
      class="c-bolt-icon__icon c-bolt-icon__icon--large"
      style=""
    >
      
          
      <use
        href="#add-open"
        style=""
      />
      
        
    </svg>
    
      
  </span>
  
    
  <!---->
</bolt-icon>
`;

exports[`<bolt-icon> Component basic usage 1`] = `
<bolt-icon name="add-open"
           size="medium"
           color="teal"
           background="square"
           aria-hidden="true"
>
  <span class="c-bolt-icon c-bolt-icon--add-open c-bolt-icon--medium c-bolt-icon--teal has-background has-square-background">
    <span class="c-bolt-icon__background-shape c-bolt-icon__background-shape--square">
    </span>
    <svg class="c-bolt-icon__icon c-bolt-icon__icon--medium"
         xmlns="http://www.w3.org/2000/svg"
         viewbox="0 0 24 24"
    >
      <g fill="none"
         fill-rule="evenodd"
      >
        <path>
        </path>
        <g fill="var(--bolt-theme-icon, currentColor)">
          <path d="M12 21a9 9 0 110-18 9 9 0 010 18m0-20a11 11 0 100 22 11 11 0 000-22M16 11h-3V8c0-.5-.4-1-1-1a1 1 0 00-1 1v3H8a1 1 0 00-1 1c0 .6.5 1 1 1h3v3c0 .6.4 1 1 1s1-.4 1-1v-3h3c.6 0 1-.4 1-1s-.4-1-1-1">
          </path>
        </g>
      </g>
    </svg>
  </span>
</bolt-icon>
`;
